packages/explorerkit-server/src/server.ts

```diff
@@ -203,49 +214,53 @@ app.post("/decode/accounts", responseDurationMiddleware, async (req: Request, re
   }
 });
 
+const decodeErrorsSchema = z.object({
+  errors: z.array(z.object({
+    programId: z.string(),
+    errorCode: z.coerce.number().nullable().optional(),
+  }).nullable()),
+});
+
+// Endpoint to decode program errors
+app.post("/decode/errors", responseDurationMiddleware, async (req: Request, res: Response) => {
+  const {data, error} = decodeErrorsSchema.safeParse(req.body);
+
+  if (error) {
+    return res.status(400).json({ error: "Invalid request body", errors: error.errors });
+  }
+
+  const programIdsWithFailure = data.errors.filter(err => err?.errorCode).map((err) => err?.programId).map(v => v!);
```

Comment:
NIT: what does this last map do? why not just `filter(v => !!v)` and then you can be sure there are no null/undefined values in the array

Reply:
It does nothing :D, I put it to make typescript linter happy as I know for sure that the value is there, but TS server still thinks programIdsWithFailure can contain null. 
There is a fix in TS 5.5 to consider filtered values: https://devblogs.microsoft.com/typescript/announcing-typescript-5-5-rc/#control-flow-narrowing-for-constant-indexed-accesses